### ISSUE TITLE
Sverchok StackExchange

### ISSUE THREAD

**Issue by DolphinDream** [nortikin/sverchok#1276]
What do you guys think about starting a stackexchange like the one blender has just for Sverchok? I prefer using a knowledge base like this over the issue feature in github. Besides such platform is geared towards this kind of things questions/answers/rewards/tagging etc and it may be more searchable. The issue tracker could still be used for developers, but for the larger user audience a stackexchange seems to be more suitable. Video tutorials are great but they lack the search ability for specific topics/how to questions. 

And taking about questions I swear I watch Jimmy showing how to color polygons in a mesh based on the distance from an empty but I can't pinpoint which one of the hundreds of videos includes this :) besides it may have been the animation nodes solution, not Sverchok I don't remember.

**Comment by enzyme69** [comment]
Isn't StackExchange already has tagging for SV and AN?

I think Sverchok and Animation Nodes together can make a beautiful book about NODES.

AN got some followers, but SV also :D Generally AN feels simpler. Not until it got to the components level. But overtime both are equally fun.

**Comment by enzyme69** [comment]
Extra info, will try:
http://stackoverflow.com/questions/17355617/can-you-add-a-light-source-in-blender-using-python

So far what I know:
- I can generate 1 lamp and many lamp, maybe give name + numbers
- I don't know how to delete and clean data in Blend properly using script, using Nodes however I have no idea

**Comment by DolphinDream** [comment]
Blender has this [Blender StackExchange](http://blender.stackexchange.com) but I feel SV / AN nodes, even though they are blender add-ons, they are entirely different animals.. and require a new way of thinking (procedurally) which is a vast domain and currently a bit esoteric. Mixing SV / AN questions with the rest of the Blender specific questions on BSE is not very effective way of building up a SV/AN knowledge base (i think), which is why I suggested SV should have its own knowledge base platform perhaps sverchok.stackexchange.com :)

**Comment by zeffii** [comment]
Bse would be great for sverchok - if people used it.
If the number of questions got large enough a case could be made for a separate exchange. questions can be migrated from A to B .  Problem is that the most interesting questions have more than one tag, and are tightly coupled with blender.

**Comment by zeffii** [comment]
I agree , videos can be hard to search and index. - usually too long and easily outdated and not editable to give better solutions when possible.

**Comment by kalwalt** [comment]
vote :+1:  for blenderexchange, And what about a forum page? [this](https://blenderartists.org/forum/showthread.php?272679-Addon-WIP-Sverchok-parametric-nodes-for-architects) is the only page i know but it is only one thread difficult to read in certain cases and not so helpful. We can also improve the wiki, it's a great possibility.

**Comment by enzyme69** [comment]
A better Sverchok documentation and more examples should be okey. Sverchok
also needs a logo :D



------------------------------------------------------------
- Jimmy Gunawan
http://blendersushi.blogspot.com.au
http://mayaspiral.blogspot.com.au
http://houdoodles.blogspot.com.au
http://puppetar.blogspot.com.au

**Comment by enzyme69** [comment]
I want to try making weekly or monthly ZINE about NODES Sverchok, Animation Nodes.

Sverchok NODE TREE "Card Example" sometimes worth it. Some kind of cards that explains the basics. 

At some point, I have made Sverchok SNIPPET at my blog, I think that's actually working. My blog was based on blogspot, it was not the best. I have Apple News blog and have few things, but Apple News is a bit closed.

My growing amount of YouTube videos, I realized ... video alone will not be enough. In the long run, the live noding video must be turned into a more compact, easy access place.

Until Sverchok version 1, and we have a more complete set of nodes and ability to talk to Animation Nodes and Blender Particle Nodes!

**Comment by zeffii** [comment]
Sverchok evolves, I prefer to spend my time writing code and docs for nodes and helping contributors evolve the code. If the nodes have good complete/ up-to-date content, then the issue is returned to the user to figure out the requirements of their system. We write docs to describe the node, but can not  include all the possible ways it could be used. 

If people don't pinpoint a missing doc / doc lacking details then we stay oblivious to the gaps. Even non coders can do a pullrequest on a single doc page, to add more info/examples -- it's not like the ways aren't already there. What we should avoid is answering case by case, questions which are really 'system of nodes' problems . "custom answers to general problems"

**Comment by enzyme69** [comment]
Watching this GIST thread:
https://github.com/nortikin/sverchok/issues/965#issuecomment-274317980

Sverchok can grow like Spore.

**Comment by zeffii** [comment]
A sobering fact about BlenderStackexchange is that people will try to give the answer that's most appropriate to the problem. Even if the poster states "I want to do X with Y" , its perfectly OK to answer with "Do X with Z instead, for these reasons".  Sometimes that means 'Sverchok' can be factored out of the equation to some degree. It can mean that using Sverchok is not necessarily the 'path of least resistance'. Other times we can answer non sverchok questions using sverchok to show how simple something can be.

Using sverchok should never be about using sverchok for the sake of using sverchok, if other methods are better go for those instead.

**Comment by DolphinDream** [comment]
Understanding each node separately is important, but also important is understanding them in context of other nodes; how they are combined and what the flow is in a system of nodes. Granted that the node documentation typically shows one node in the context of others, but it obviously does not extend too far past the basic demo illustrating the node’s capabilities. A stackexcange in which you can get various custom solutions to a given specific problem it still a nice to have resource.

**Comment by DolphinDream** [comment]
I don’t mind posting questions on BSE if that works, I only worry they will be buried under a lot of Blender specific questions and never really get the answer I’m looking for. I don’t know what the SV user base is and if that is large enough for people to answer or for other SV users to be interested in finding an answer. At the beginning I imagine that the developers and the advance users would be the ones posting questions and answers and perhaps as the knowledge base grows more users will join in to learn SV and to contribute to the knowledge base.

**Comment by zeffii** [comment]
The reason I prefer not to field general questions here on the issue tracker ( even tho we have done so.. a lot...and it has been encouraged )  is because i think BSE would promote more users to answer, because it has a much much larger viewing public. Good parametric questions will get great answers there are smart people there who use Sverchok but don't participate on the tracker at all.

**Comment by zeffii** [comment]
>  I only worry they will be buried under a lot of Blender specific questions and never really get the answer I’m looking for.

In that case, we can start a single thread on here  where you can post a new BSE question links.. and if I or anyone else reading has time, we can head over to BSE to answer it. (I've turned off any notification from BSE for the time being.. )

If more people see the sverchok tag on BSE, it may spike their interest too. Doing everything here has proven a little insular.

**Comment by zeffii** [comment]
I get it tho.. a dedicated exchange for Sverchok would be neat. I agree it's its own little system in many ways, but the strength is Sverchok is really that it operates in the larger Blender ecosystem.

**Comment by DolphinDream** [comment]
I agree BSE would be a good option to increase interest and keep things less insular. If we could get to link the SV BSE postings so that SV devs and advanced SV users can answer them until the SV user base grows enough to not need direct SV/Dev babysitting that would be great.

**Comment by zeffii** [comment]
i think BSE can set notifications to alert you when a question is tagged `sverchok`

**Comment by nortikin** [comment]
i remember IRC #
